### PR TITLE
Android 6: App crash when read Contact permission is granted (#2064)

### PR DESCRIPTION
**Pull request by bmarty**
Ugly fix for #2064.
Tested ok on Emulator running Android 6 (API 23).

I cannot find a better fix for now. @onurays @BillCarsonFr @ganfra any thoughts?

### CODE REVIEW

File: vector/src/main/java/im/vector/app/core/platform/VectorBaseActivity.kt [hunk 1]
```diff
@@ -345,6 +345,14 @@ abstract class VectorBaseActivity : AppCompatActivity(), HasScreenInjector {
                 }
     }
 
+    protected var postResumeScheduledAction: Runnable? = null
```
Comment (ganfra) [inline vector/src/main/java/im/vector/app/core/platform/VectorBaseActivity.kt:348]: Could u create a list instead? so we can handle multiple things

File: vector/src/main/java/im/vector/app/features/createdirect/CreateDirectRoomActivity.kt [hunk 2]
```diff
@@ -113,7 +115,13 @@ class CreateDirectRoomActivity : SimpleFragmentActivity() {
     override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
         if (allGranted(grantResults)) {
             if (requestCode == PERMISSION_REQUEST_CODE_READ_CONTACTS) {
-                addFragmentToBackstack(R.id.container, ContactsBookFragment::class.java)
+                try {
+                    addFragmentToBackstack(R.id.container, ContactsBookFragment::class.java)
```
Comment (ganfra) [inline vector/src/main/java/im/vector/app/features/createdirect/CreateDirectRoomActivity.kt:119]: I wouldn't even try to do this, directly post it to postResume?

**Review by ganfra — CHANGES_REQUESTED** [review]
Some remarks. Also, you might want to check for other fragment transactions

File: vector/src/main/java/im/vector/app/features/createdirect/CreateDirectRoomActivity.kt (hunk 2 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/app/features/createdirect/CreateDirectRoomActivity.kt:119]: the bug is only for Android 6, so I assume it is a workaround... But Ok I will change that

File: vector/src/main/java/im/vector/app/core/platform/VectorBaseActivity.kt (hunk 1 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/app/core/platform/VectorBaseActivity.kt:348]: No usage for the moment (so no need to do complicated things), but yes, why not...

File: vector/src/main/java/im/vector/app/features/createdirect/CreateDirectRoomActivity.kt (hunk 2 above)
Comment (ganfra) [reply vector/src/main/java/im/vector/app/features/createdirect/CreateDirectRoomActivity.kt:119]: It's more a question than a request, I don't know if it'll work

**Review by ganfra — APPROVED** [review]
Thanks for the changes, LGTM now, but I did not test it

**Comment by bmarty** [comment]
I tested emulator API 23 and real device API 29

**Pull request merged**